OpenEdXMobile/src/main/java/org/edx/mobile/module/db/impl/DbHelper.java

```diff
@@ -77,73 +80,86 @@ private void createAssessmentTable(SQLiteDatabase db){
+        if (oldVersion < 6) {
+            db.beginTransaction();
+            try {
+                final File externalAppDir = FileUtil.getExternalAppDir(context);
+                final String previousAppDirPath = TextUtils.join("/", Arrays.<CharSequence>asList(
+                        Environment.getExternalStorageDirectory().getAbsolutePath(),
+                        "Android", "data", context.getPackageName())).toString();
+                if (externalAppDir != null) {
                     Cursor cursor = db.query(false, DbStructure.Table.DOWNLOADS,
                             new String[]{DbStructure.Column.ID, DbStructure.Column.USERNAME,
                                     DbStructure.Column.FILEPATH}, null, null, null, null, null, null);
                     if (cursor != null) {
                         try {
+                            final int idIndex = cursor.getColumnIndexOrThrow(DbStructure.Column.ID);
+                            final int usernameIndex = cursor.getColumnIndexOrThrow(DbStructure.Column.USERNAME);
+                            final int filePathIndex = cursor.getColumnIndexOrThrow(DbStructure.Column.FILEPATH);
+
                             while (cursor.moveToNext()) {
-                                final String id = cursor.getString(0);
-                                final String username = cursor.getString(1);
-                                final String filePath = cursor.getString(2);
+                                final String id = cursor.getString(idIndex);
+                                final String username = cursor.getString(usernameIndex);
+                                final String filePath = cursor.getString(filePathIndex);
                                 final String hashedUsername = Sha1Util.SHA1(username);
-                                final String newFilePath = filePath.replace(
-                                        appExternalDir.getAbsolutePath() + "/" + username,
-                                        appExternalDir.getAbsolutePath() + "/"
-                                                + AppConstants.Directories.VIDEOS + "/"
-                                                + hashedUsername);
-
-                                // First update the directory name along with its whole path
-                                final File previousDir = new File(appExternalDir, username);
+                                final String newFilePath = filePath.replaceFirst(
+                                        "^" + TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                previousAppDirPath, username)),
+                                        TextUtils.join("/", Arrays.<CharSequence>asList(
+                                                externalAppDir.getAbsolutePath(),
+                                                AppConstants.Directories.VIDEOS,
+                                                hashedUsername)).toString());
```

Comment:
Hmm, OK. Which reminds me that you should also change the `previousDir` path to use `previousAppDirPath` instead of `externalAppDir`.